planning/mission_planner/src/lanelet2_plugins/default_planner.cpp

```diff
@@ -92,6 +93,42 @@ bool is_in_parking_lot(
   return false;
 }
 
+bool is_footprint_in_lanelet(
+  const lanelet::ConstLanelet & goal_lanelet, tier4_autoware_utils::LinearRing2d goal_footprint,
+  lanelet::routing::RoutingGraphPtr routing_graph_ptr_, lanelet::ConstLanelets path_lanelets)
+{
+  std::vector<tier4_autoware_utils::Point2d> points_intersection;
+
+  // check if goal footprint is in current lane
+  boost::geometry::intersection(
+    goal_footprint, goal_lanelet.polygon2d().basicPolygon(), points_intersection);
+  if (points_intersection.empty()) {
+    return true;
+  }
+  points_intersection.clear();
+
+  // check if goal footprint is in between many lanelets
+  lanelet::ConstLanelet combined_prev_lanelet;
+  combined_prev_lanelet = combine_lanelets(path_lanelets);
+  auto next_lanelets = routing_graph_ptr_->following(goal_lanelet);
+  for (const auto & next_lane : next_lanelets) {
+    lanelet::ConstLanelets lanelets;
+    lanelets.push_back(combined_prev_lanelet);
+    lanelets.push_back(next_lane);
+
+    auto combined_lanelets = combine_lanelets(lanelets);
+    boost::geometry::intersection(
+      goal_footprint, combined_lanelets.polygon2d().basicPolygon(), points_intersection);
+
+    if (points_intersection.empty()) {
+      return true;
+    }
+    points_intersection.clear();
+  }
```

Comment:
Here, only next lanelets are considered.
When lanelets longitudinal length is too short, not only the next lanelets but also the lanelet after the next, ... must be considered, right?

Reply:
Thanks for your review, you are right, I forgot concerning about next lanelets of goal. I will think of a suitable approach for this situation.